src/main/scala/mesosphere/marathon/tasks/TaskIDUtil.scala

```diff
@@ -17,8 +17,10 @@ class TaskIdUtil {
     appId.safePath + taskDelimiter + uuidGenerator.generate()
   }
 
-  def appId(taskId: TaskID): PathId = {
-    taskId.getValue match {
+  def appId(taskId: TaskID): PathId = appId(taskId.getValue)
+
+  def appId(taskId: String): PathId = {
+    taskId match {
```

Comment:
Please handle the failing case explicitly, since that means you have to handle a MatchError